### DISCUSSION TITLE
Directories synchronization

### DISCUSSION THREAD

**Discussion by mikekazakov** [mikekazakov/nimble-commander#235] [Ideas]
This is a "must have" functionality, and NC is still lacking it.
Have to do some research first.

**Comment by mikekazakov** [comment]
- compare directories.

**Comment by gingerbeardman** [comment]
I just put this comparison together for fman and thought I'd post it here, too.

I often need to compare directories, either to simply spot differences (new/missing file) or to sync them by subsequently copying the extra files to another location. An app can help me by selecting files in one/both panes based on a comparison.

> Looking at the Nimble way of doing things, perhaps a command similar to "Select by Mask..." maybe named "Select by Comparison..." which does the user configured comparison. I doubt the comparison has to be set each time, so that could be brought up if Option is held whilst choosing the command? It could be as easy as parsing the output of `diff pane1 pane2`?

Other dual pane apps that do it in a variety of ways...

- [XYplorer](https://www.xyplorer.com/) (Windows) does it using operators on the select function:
![xy1](https://user-images.githubusercontent.com/49612/43214906-a1a015e2-9032-11e8-8830-80c40b7aa2a1.png)
![xy2](https://user-images.githubusercontent.com/49612/43214907-a1b61d56-9032-11e8-9175-2a3d13280ad7.png)

- [Nomad](http://www.nomad-net.info) (Windows) does it in a similar way, but with a more complicated dialog:
![nomad1](https://user-images.githubusercontent.com/49612/43214910-a3fbdde4-9032-11e8-9c77-61bde71eae7c.png)
![nomad2](https://user-images.githubusercontent.com/49612/43214911-a4180884-9032-11e8-9546-c0a9bbfb9d93.png)

- [ForkLift](https://binarynights.com) (Mac) has a sync mode that will show you the files, but you can't select them:
![screen shot 2018-07-25 at 19 54 44](https://user-images.githubusercontent.com/49612/43222468-1b3bb806-9048-11e8-9497-67f42d3e4216.png)

- [Double Commander](https://doublecmd.sourceforge.io) (All) is the same: sync mode will show you the files, but you can't select them:
![screen shot 2018-07-25 at 21 13 52](https://user-images.githubusercontent.com/49612/43225137-53d35a36-9050-11e8-8b69-81916e057055.png)

- [fman](https://fman.io) (All ) doesn't have it, but I've requested it: https://github.com/fman-users/fman/issues/454

**Comment by M0rtenB** [comment]
Any news on this? Would be banger'

**Comment by mikekazakov** [comment]
No news and it doesn't seem likely that I'll work on this feature anytime soon.
If anybody is interested enough - PRs are welcome.
